# SPDX-FileCopyrightText: Copyright (c) 2025 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
# SPDX-License-Identifier: Apache-2.0

"""Workmesh protobuf messages."""

from workmesh.messages.messages_pb2 import (
    # Messages
    Animation,
    AnimationData,
    AnimationFrame,
    AnimationProperties,
    Audio,
    AudioProperties,
    BoundingBox,
    ChangeVolume,
    ClipData,
    ClipProperties,
    ClipStatus,
    Color,
    Command,
    EulerAngle,
    FilePublish,
    FillCircleAnimation,
    Frame,
    GenerationType,
    HumanSpeechRequest,
    # Enums
    ImageEncoding,
    ImageGeneration,
    Keypoint,
    LightCommand,
    LookAtParameters,
    Ping,
    PlayClip,
    Pong,
    Position2D,
    PresenceStatus,
    ProceduralClip,
    ProceduralClipPause,
    ProceduralClipStart,
    ProceduralClipStop,
    ProceduralClipUpdate,
    ProceduralState,
    ProceduralType,
    ProcessedDoA,
    RealtimeDoA,
    RemoteControlCommand,
    Robot,
    RobotFrame,
    RobotSpeechRequest,
    Service,
    ServiceCommand,
    Skeleton,
    StaticAnimation,
    StopClip,
    ToolStatus,
    TrackingStatus,
    TrackParameters,
    Transition,
    UserDetection,
    UserState,
    UserTrackingStatus,
    UserUtterance,
    UserUtteranceStatus,
)

__all__ = [
    # Enums
    "ImageEncoding",
    "GenerationType",
    "Color",
    "ProceduralType",
    "ProceduralState",
    "Robot",
    "Service",
    "Command",
    "PresenceStatus",
    "TrackingStatus",
    "UserUtteranceStatus",
    # Messages
    "Animation",
    "AnimationData",
    "AnimationFrame",
    "EulerAngle",
    "Audio",
    "PlayClip",
    "AnimationProperties",
    "AudioProperties",
    "ClipProperties",
    "ClipData",
    "StopClip",
    "ChangeVolume",
    "Transition",
    "BoundingBox",
    "ClipStatus",
    "RealtimeDoA",
    "ProcessedDoA",
    "FilePublish",
    "Frame",
    "HumanSpeechRequest",
    "ImageGeneration",
    "StaticAnimation",
    "FillCircleAnimation",
    "LightCommand",
    "Ping",
    "Pong",
    "Position2D",
    "ProceduralClip",
    "ProceduralClipStart",
    "ProceduralClipStop",
    "ProceduralClipPause",
    "ProceduralClipUpdate",
    "LookAtParameters",
    "TrackParameters",
    "RemoteControlCommand",
    "RobotFrame",
    "RobotSpeechRequest",
    "ServiceCommand",
    "ToolStatus",
    "Keypoint",
    "Skeleton",
    "UserDetection",
    "UserState",
    "UserTrackingStatus",
    "UserUtterance",
]
